tests/inc/class-test-admin-scripts.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Test functions in the Admin_Scripts\Admin namespace.
+ */
+
+namespace Asset_Loader\Tests;
+
+use Asset_Loader\Admin;
+use WP_Mock;
+
+class Test_Admin_Scripts extends Asset_Loader_Test_Case {
+	/**
+	 * Test maybe_set_ssl_cert_error_handling().
+	 *
+	 * @dataProvider provide_maybe_setup_ssl_cert_error_handling_cases
+	 */
+	public function test_maybe_setup_ssl_cert_error_handling( bool $is_admin, string $script_uri, bool $expect_action, string $message ) : void {
+		WP_Mock::userFunction( 'is_admin' )->andReturn( $is_admin );
+		if ( $expect_action ) {
```

Comment:
This is one of the examples where Brain Monkey provides some more useful helpers. It stubs the `has_action` function (using a fake registry), so your code here could be as simple as this:

```php
$this->assertSame( $should_have_added_action, has_action( ... ) );
```

No need for a conditional and/or two different functions to check what you're interested in.

Reply:
That sounds really appealing; swapping mock library's out of scope right now, but i'll look into it for later on@